### PR TITLE
Fix codim-1 assembly on interior facets

### PR DESCRIPTION
**Pull request by jorgensd**
There were several issues with codim-1 meshes combined with interior facet meshes.

1. The entity maps were not used when computing the domain for the relevant coefficient (for packing).
2. The element tables in FFCx had quadrature permutations, even if the entity should be consistently oriented due to https://github.com/FEniCS/dolfinx/pull/3209

This PR is coupled with https://github.com/FEniCS/ffcx/pull/717 and adds a check for both scalar and vector assembly that fails on main.

Thanks to @jpdean for being a good sparring partner on getting into the details here

### CODE REVIEW

**Comment by jorgensd** [comment]
I've organized the commits such that:
Commit 1: f7ccb449ce62e8157c3e7b8e1c84ee555383b306  fails on assemble_scalar comparison: https://github.com/FEniCS/dolfinx/actions/runs/11163689143/job/31031263806
Commit 2: e18ba7d42881111bf0d4a8640e68d189788432a0: fails with assemble_vector: https://github.com/FEniCS/dolfinx/actions/runs/11163698693/job/31031293844
Commit 3: 03d45a575c742e661e34d369a6a1dc32a8aca24c passes: https://github.com/FEniCS/dolfinx/actions/runs/11163840864/job/31031762628?pr=3452

File: python/test/unit/fem/test_assemble_submesh.py [hunk 1]
```diff
@@ -609,3 +610,73 @@ def test_mixed_measures():
...
+    F_compiled = fem.form(F, entity_maps=entity_maps)
+    J_compiled = fem.form(J, entity_maps=entity_maps)
+    J_global = msh.comm.allreduce(fem.assemble_scalar(J_compiled), op=MPI.SUM)
+    b = fem.assemble_vector(F_compiled)
+    b.scatter_reverse(la.InsertMode.add)
+    b.scatter_forward()
+
+    x = ufl.SpatialCoordinate(msh)
+    J_ref = fem.form(ufl.avg(f(x)) * ufl.dS(metadata=metadata))
+    F_ref = fem.form(ufl.inner(ufl.jump(v), f(x)) * dS)
```
Comment (nate-sime) [inline python/test/unit/fem/test_assemble_submesh.py:674]: Shouldn't this be `ufl.dS(metadata=metadata)`?

**Review by nate-sime — APPROVED** [review]

File: .github/workflows/ccpp.yml [hunk 2]
```diff
@@ -104,7 +104,7 @@ jobs:
         run: |
           pip install git+https://github.com/FEniCS/ufl.git
           pip install git+https://github.com/FEniCS/basix.git
-          pip install git+https://github.com/FEniCS/ffcx.git
+          pip install git+https://github.com/FEniCS/ffcx.git@dokken/codim-1-interior-facet
```
Comment (jorgensd) [inline .github/workflows/ccpp.yml:107]:
```suggestion
          pip install git+https://github.com/FEniCS/ffcx.git
```

File: .github/workflows/ccpp.yml [hunk 3]
```diff
@@ -174,7 +174,7 @@ jobs:
...
```
Comment (jorgensd) [inline .github/workflows/ccpp.yml:177]:
```suggestion
          pip install git+https://github.com/FEniCS/ffcx.git
```

File: .github/workflows/ccpp.yml [hunk 4]
```diff
@@ -252,7 +252,7 @@ jobs:
...
```
Comment (jorgensd) [inline .github/workflows/ccpp.yml:255]:
```suggestion
          pip install git+https://github.com/FEniCS/ffcx.git
```

File: .github/workflows/macos.yml [hunk 5]
```diff
@@ -64,7 +64,7 @@ jobs:
         run: |
           python -m pip install git+https://github.com/FEniCS/basix.git
           python -m pip install git+https://github.com/FEniCS/ufl.git
-          python -m pip install git+https://github.com/FEniCS/ffcx.git
+          python -m pip install git+https://github.com/FEniCS/ffcx.git@dokken/codim-1-interior-facet
```
Comment (jorgensd) [inline .github/workflows/macos.yml:67]:
```suggestion
          python -m pip install git+https://github.com/FEniCS/ffcx.git
```

File: .github/workflows/oneapi.yml [hunk 6]
```diff
@@ -59,7 +59,7 @@ jobs:
         with:
           path: ./ffcx
           repository: FEniCS/ffcx
-          ref: main
+          ref: dokken/codim-1-interior-facet
```
Comment (jorgensd) [inline .github/workflows/oneapi.yml:62]:
```suggestion
          ref: main
```

File: .github/workflows/pyvista.yml [hunk 7]
```diff
@@ -40,7 +40,7 @@ jobs:
         run: |
           pip install git+https://github.com/FEniCS/ufl.git
           pip install git+https://github.com/FEniCS/basix.git
-          pip install git+https://github.com/FEniCS/ffcx.git
+          pip install git+https://github.com/FEniCS/ffcx.git@dokken/codim-1-interior-facet
```
Comment (jorgensd) [inline .github/workflows/pyvista.yml:43]:
```suggestion
          pip install git+https://github.com/FEniCS/ffcx.git
```

File: .github/workflows/redhat.yml [hunk 8]
```diff
@@ -32,7 +32,7 @@ jobs:
         run: |
           python3 -m pip install git+https://github.com/FEniCS/ufl.git
           python3 -m pip install git+https://github.com/FEniCS/basix.git
-          python3 -m pip install git+https://github.com/FEniCS/ffcx.git
+          python3 -m pip install git+https://github.com/FEniCS/ffcx.git@dokken/codim-1-interior-facet
```
Comment (jorgensd) [inline .github/workflows/redhat.yml:35]:
```suggestion
          python3 -m pip install git+https://github.com/FEniCS/ffcx.git
```

File: .github/workflows/sonarcloud.yml [hunk 9]
```diff
@@ -60,7 +60,7 @@ jobs:
         run: |
           python3 -m pip install git+https://github.com/FEniCS/ufl.git
           python3 -m pip install git+https://github.com/FEniCS/basix.git
-          python3 -m pip install git+https://github.com/FEniCS/ffcx.git
+          python3 -m pip install git+https://github.com/FEniCS/ffcx.git@dokken/codim-1-interior-facet
```
Comment (jorgensd) [inline .github/workflows/sonarcloud.yml:63]:
```suggestion
          python3 -m pip install git+https://github.com/FEniCS/ffcx.git
```

**Pull request merged**